packages/cli/src/webhooks/webhook.service.ts

```diff
@@ -118,4 +133,212 @@ export class WebhookService {
 			.find({ select: ['method'], where: { webhookPath: path } })
 			.then((rows) => rows.map((r) => r.method));
 	}
+
+	async createWebhookIfNotExists(
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		mode: WorkflowExecuteMode,
+		activation: WorkflowActivateMode,
+	): Promise<void> {
+		const webhookExists = await this.runWebhookMethod(
+			'checkExists',
+			workflow,
+			webhookData,
+			mode,
+			activation,
+		);
+		if (!webhookExists) {
+			// If webhook does not exist yet create it
+			await this.runWebhookMethod('create', workflow, webhookData, mode, activation);
+		}
+	}
+
+	async deleteWebhook(
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		mode: WorkflowExecuteMode,
+		activation: WorkflowActivateMode,
+	) {
+		await this.runWebhookMethod('delete', workflow, webhookData, mode, activation);
+	}
+
+	private async runWebhookMethod(
+		method: WebhookSetupMethodNames,
+		workflow: Workflow,
+		webhookData: IWebhookData,
+		mode: WorkflowExecuteMode,
+		activation: WorkflowActivateMode,
+	): Promise<boolean | undefined> {
+		const node = workflow.getNode(webhookData.node);
+
+		if (!node) return;
+
+		const nodeType = this.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
+
+		const webhookFn = nodeType.webhookMethods?.[webhookData.webhookDescription.name]?.[method];
+		if (webhookFn === undefined) return;
+
+		const context = new HookContext(
+			workflow,
+			node,
+			webhookData.workflowExecuteAdditionalData,
+			mode,
+			activation,
+			webhookData,
+		);
+
+		return (await webhookFn.call(context)) as boolean;
```

Comment:
it's an issue in out tsconfig that is picking up very old typings for `.call` from ES5 
![image](https://github.com/user-attachments/assets/7bd996df-0e03-452a-8473-2dec95753aa0)

Reply:
This will get fixed once we switch on `strictBindCallApply` in tsconfig. When we do, `@typescript-eslint/no-unnecessary-type-assertion` will start failing, and we can then clean these up.